frontend/src/metabase-lib/expressions/diagnostics.ts

```diff
@@ -151,8 +163,17 @@ function prattCompiler(source, startRule, legacyQuery, name) {
     }
   } catch (err) {
     console.warn("compile error", err);
-    return err;
+    return err as ErrorWithMessage;
   }
 
   return null;
 }
+
+function isBooleanExpression(
+  expr: unknown,
+): expr is [string, ...Expr[]] & { node: Node } {
```

Comment:
I tried `1 + 1` expression and there is no `node` attribute, but this type guard says there is.

Reply:
made node optional